burn-core/src/module/param/tensor.rs

```diff
@@ -4,22 +4,36 @@ use crate::tensor::{
     backend::{AutodiffBackend, Backend},
     Tensor,
 };
+use alloc::vec::Vec;
+use burn_tensor::{Bool, Int};
 
 impl<B: Backend, const D: usize> From<Tensor<B, D>> for Param<Tensor<B, D>> {
     fn from(value: Tensor<B, D>) -> Self {
         Param::new(ParamId::new(), value.require_grad())
     }
 }
 
```

Comment:
Do we need to explain why we do not require grad?

Reply:
I added why we add require grad, other tensor (bool and int) can't have gradients